libbeat/publisher/queue/spool/spool.go

```diff
@@ -76,8 +76,8 @@ type Settings struct {
 const minInFlushTimeout = 100 * time.Millisecond
 const minOutFlushTimeout = 0 * time.Millisecond
 
-// NewSpool creates and initializes a new file based queue.
-func NewSpool(logger logger, path string, settings Settings) (*Spool, error) {
+// newDiskSpool creates and initializes a new file based queue.
+func newDiskSpool(logger logger, path string, settings settings) (*diskSpool, error) {
```

Comment:
These have been public on purpose, so to allow us to reuse the Spool as is as a library. The 'Factory' approach to generate a queue.Queue would require us to go with an untyped config.
This might become interesting for the docker logging plugin in the future, maybe, maybe not :)

Reply:
Well, `memqueue.NewQueue` is exported for similar reasons (it is used in only one place to create an explicit queue with hard-coded settings). But I suspect the need to create a transactional on-disk queue is relatively rare -- at least, it wouldn't be a lightweight or casual choice -- so it's easy to revisit the exported API once such an application does arise.

My preference, since this feature is beta and other planned uses don't yet have solid requirements, is to start with things internal and make elements public as we're confident about the interface we want to support in the future. But if you feel strongly I can re-export `Settings` and wrap this helper in `spool.NewQueue` as in the memqueue case, which would at least restrict external dependencies to the queue interface instead of the specific implementing type.